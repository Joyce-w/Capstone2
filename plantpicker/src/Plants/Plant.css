.Plant-div{
    display: flex;
    flex:1;
    margin: auto 10%;
    margin-top:4%;
}

h1{
    text-align: center;
}

.Plant-leftDiv{
    padding: 2em;
    margin: 0 2em;
    width:50%;
}

.Plant-rightDiv{
    width:60%;
    padding: 1em 3em;
}

img{
    margin:2em 0;
    width: 35em;
}

/* regular select menu */
select{
    border-radius: 10px;
}
button{
    background: none;
    border: none;
}
/* Drop down menu */
.dropbtn {
    background-color: #3498DB;
    color: white;
    padding: 16px;
    font-size: 16px;
    border: none;
    cursor: pointer;
}

.dropbtn:hover,
.dropbtn:focus {
    background-color: #2980B9;
}

.dropdown {
    position: relative;
    display: inline-block;
}

.dropdown-content {
    display: none;
    position: absolute;
    background-color: #f1f1f1;
    min-width: 160px;
    overflow: auto;
    box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
    z-index: 1;
}

.dropdown-content p {
    color: black;
    padding: 12px 16px;
    text-decoration: none;
    display: block;
}

.dropdown p:hover {
    background-color: #ddd;
}

.show {
    display: block;
}





/**MEDIA QUERIES**/
@media screen and (max-width: 1024px) {
    .Plant-div {
        flex-direction: column;
        margin: none;
    }
    .Plant-leftDiv {
        width:80%;
        position: relative;
        margin: 0 0;
    }
    img {
        width: 80%;
    }
    
    .Plant-rightDiv {
        padding: 2em;
        margin: 0 0;
        width:80%;
    }
}