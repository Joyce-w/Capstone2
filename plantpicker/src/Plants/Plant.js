import React, { useState, useEffect, useRef } from "react";
import { useParams } from "react-router-dom";
import { PawPrint, Baby, Wind, Drop, PlusCircle } from "phosphor-react";
import "./Plant.css";
import PlantsApi from "../api";
import { decodeToken } from "react-jwt";


function Plant() {

    const { plant_name } = useParams();
    const [water, setWater] = useState("");
    
    //create state to store plant data
    const [plant, setPlant] = useState({});
    //Get all plants data
    useEffect(() => {
        async function getPlant(name) {
            
            const res = await PlantsApi.getPlant(name);
            setPlant(() => res);
        }

        getPlant(plant_name)
        
        //check value of drought tolerance of plant
        if (plant.drought_tolerant === "low") {
            setWater("Soil should be constantly wet but not soaking")
        } else if (plant.drought_tolerant === "medium") {
            setWater("Can go a few days without watering.")
        } else {
            setWater("Can go a week without watering!")
        }
    }, [plant_name])
    
    // Get user plant list
    const [user, setUser] = useState([])
  
    //retrieve token from local storage and decode
    let userToken = decodeToken(localStorage.getItem('token')) || null;
    
    useEffect(() => {
        async function getUser(currUser) {
            const res = await PlantsApi.getUser(currUser);
            setUser(res.plant_lists)
        }
        getUser(userToken.username)

    }, [])

    const dropDownRef = useRef();
    const listRef = useRef();
    const toggleDropdown = (e) => {
        console.log(e)
        if (e.target.className === 'dropbtn') {
            console.log('whats next')
            let dropdown = dropDownRef.current.className;

            for (let i = 0; i < dropdown.length; i++) {
            var openDropdown = dropdown[i];
            if (openDropdown.classList.contains('show')) {
                openDropdown.classList.remove('show');
            }
            }
        }
    }

    //handle add to list
    // const addPlant = (e) => {
    //     e.preventDefault();
    //     console.log(e)
    // }


    return (
        <>
        <div className="Plant-div">
            
            <div className="Plant-leftDiv">
                <img src={plant.img} alt={plant.plant_name}></img>
            </div>
                <div className="Plant-rightDiv">
                    <h1>{plant.plant_name}</h1>
                    <p>{plant.details}</p>
                    <p>Max Height: { plant.max_height}"</p>
                    <p>Temperature Range: {plant.min_temp}-{plant.max_temp} °F |  °C</p>
                    <p>Produces Flowers: {plant.flowering ? "Yes" : "No"}</p>
                    
                    <hr></hr>
                    
                    <h4>Additional Notes</h4>
                    <p className="pet-friendly"><PawPrint size={20} />{ plant.pet_friendly ? "Pet friendly" : "Can be toxic to pets!"}</p>
                    <p className="kid-friendly"><Baby size={20} />{ plant.kid_friendly ? "Kid friendly" : "Take caution around children"}</p>
                    <p className="air-purify"><Wind size={20} />{ plant.air_purifying ? "Air purifying" : "Not as effective in purifying the air but still nice to have around!"}</p>
                    <p className="watering"><Drop size={20} /> {water} </p>
                    <br></br>

                    {/* Select user list to add */}
                    

                    <label for="plant-list"> Add to List </label>
                    <select id="plant-list" name="plant-list" form="plantList">
                        {/* <option> Add to List</option> */}
                            {user.map(list => <option value={ list.list_id}>{list.list_name}</option> )}
                    </select>
                    <button > <PlusCircle size={30} /></button>



                </div>
                
        </div>
        </>
    )
}

export default Plant;