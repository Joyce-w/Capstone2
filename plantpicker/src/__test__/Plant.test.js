import { render, fireEvent, screen } from "@testing-library/react";
import {MemoryRouter} from 'react-router-dom'
import Plant from "../Plants/Plant";

it("renders w/o crashing", () => {
    render(<MemoryRouter><Plant/></MemoryRouter>)
})

it("matches snapshot", function () {
    const { asFragment } = render(<MemoryRouter><Plant /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
})

test("if static text is on the page", () => {
    const { getByText,debug } = render(<MemoryRouter><Plant /></MemoryRouter>);

    let height = getByText("Max Height");
    let notes = getByText("Additional Notes");

    expect(height).toBeInTheDocument();
    expect(notes).toBeInTheDocument();
})
