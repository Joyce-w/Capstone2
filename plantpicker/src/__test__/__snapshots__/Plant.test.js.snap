// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Plant-div"
  >
    <div
      class="Plant-leftDiv"
    >
      <img />
    </div>
    <div
      class="Plant-rightDiv"
    >
      <h1 />
      <p />
      <p>
        Max Height: "
      </p>
      <p>
        Temperature Range: - °F |  °C
      </p>
      <p>
        Produces Flowers: No
      </p>
      <hr />
      <h4>
        Additional Notes
      </h4>
      <p
        class="pet-friendly"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="212"
            cy="108"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="44"
            cy="108"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="92"
            cy="60"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="164"
            cy="60"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M183.23572,155.85352a43.541,43.541,0,0,1-20.667-25.90381l-.00092.001a35.9985,35.9985,0,0,0-69.13562,0l-.00092-.001a43.541,43.541,0,0,1-20.667,25.90381,32.00736,32.00736,0,0,0,27.72266,57.61767,72.51938,72.51938,0,0,1,55.02612,0,32.00679,32.00679,0,0,0,27.72266-57.61767Z"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        Can be toxic to pets!
      </p>
      <p
        class="kid-friendly"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="96"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="92"
            cy="128"
            r="12"
          />
          <circle
            cx="164"
            cy="128"
            r="12"
          />
          <path
            d="M154.53832,168.00269a48.03112,48.03112,0,0,1-53.07491.00115"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M128,32c-16,20-16,40-16,40a16,16,0,0,0,32,0"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        Take caution around children
      </p>
      <p
        class="air-purify"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M129.48449,192.32854A24.00285,24.00285,0,1,0,152,160H40"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M97.48449,63.67146A24.00285,24.00285,0,1,1,120,96H24"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M185.48449,95.67146A24.00285,24.00285,0,1,1,208,128H32"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        Not as effective in purifying the air but still nice to have around!
      </p>
      <p
        class="watering"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M208,144c0-72-80-128-80-128S48,72,48,144a80,80,0,0,0,160,0Z"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M136.11987,191.216a47.923,47.923,0,0,0,39.20472-39.1564"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
         Can go a week without watering! 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
